ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/infrastructure/analyzers/directoryAnalyzers/DirectoryAnalyzerByText.java

```diff
@@ -0,0 +1,43 @@
+package edu.usb.argos.ASTProcessor.infrastructure.analyzers.directoryAnalyzers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+import edu.usb.argos.ASTProcessor.application.analyzers.IDirectoryAnalyzer;
+import edu.usb.argos.ASTProcessor.application.analyzers.IFileAnalyzer;
+import edu.usb.argos.ASTProcessor.application.exceptions.FileAnalyzerException;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Component
+@AllArgsConstructor
+public class DirectoryAnalyzerByText<TAst> implements IDirectoryAnalyzer<String[], TAst> {
+
+    private final IFileAnalyzer<String, TAst> fileAnalyzer;
+
+    @Override
+    public List<TAst> analyzeDirectory(String[] sourceCode) {
+        List<TAst> astFiles = new ArrayList<>();
+        TAst ast;
+
+        for (String code : sourceCode) {
+
+            try {
+                ast = fileAnalyzer.readFile(code);
+                if (ast != null) {
+                    astFiles.add(ast);
+                }
+            } catch (FileAnalyzerException exception) {
+                String errorMessage = "Error getting AST from the source";
+                log.error(errorMessage, exception);
```

Comment:
it is necessary to throw and specific exception here

Reply:
Done